nixos/modules/services/monitoring/gatus.nix

```diff
@@ -0,0 +1,144 @@
+    enable = mkEnableOption "Gatus";
+
+    package = mkPackageOption pkgs "gatus" { };
+
+    user = mkOption {
+      type = str;
+      default = "gatus";
+      description = "User account under which Gatus runs.";
+    };
+
+    group = mkOption {
+      type = str;
+      default = "gatus";
+      description = "Group under which Gatus runs.";
+    };
+
+    configFile = mkOption {
+      type = path;
+      default = settingsFormat.generate "gatus.yaml" cfg.settings;
+      description = ''
+        Path to the Gatus configuration file.
+        Overrides any configuration made using the `settings` option.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = nullOr path;
+      default = null;
+      description = ''
+        File to load as environment file.
+        Environmental variables from this file can be interpolated in the configuration file using `''${VARIABLE}`.
+        This is useful to avoid putting secrets into the nix store.
+      '';
+    };
+
+    settings = mkOption {
+      type = submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          web.port = mkOption {
+            type = int;
+            default = 8080;
+            description = ''
+              The TCP port to serve the Gatus service at.
+            '';
+          };
+        };
+      };
+
+      default = { };
+
+      example = literalExpression ''
+        {
+          web.port = 8080;
+          endpoints = [{
+            name = "website";
+            url = "https://twin.sh/health";
+            interval = "5m";
+            conditions = [
+              "[STATUS] == 200"
+              "[BODY].status == UP"
+              "[RESPONSE_TIME] < 300"
+            ];
+          }];
+        }
+      '';
+
+      description = ''
+        Configuration for Gatus.
+        Supported options can be found at the [docs](https://gatus.io/docs).
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Whether to open the firewall for the Gatus web interface.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.gatus = {
+      description = "Automated developer-oriented status page";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        Restart = "unless-stopped";
+        ExecStart = getExe cfg.package;
+        StateDirectory = "gatus";
+        SyslogIdentifier = "gatus";
+        EnvironmentFile = lib.optional (cfg.environmentFile != null) cfg.environmentFile;
+      };
+
+      environment = {
+        GATUS_CONFIG_PATH = cfg.configFile;
+      };
+    };
+
+    users.users.gatus = mkIf (cfg.user == "gatus") {
+      isSystemUser = true;
+      group = cfg.group;
+      home = "/var/lib/${config.systemd.services.gatus.serviceConfig.StateDirectory}";
+    };
```

Comment:
Would it be possible to use a dynamic user instead?

Reply:
I didn't know of this feature, I think this will work.

Reply:
Better and better at each iteration !

Reply:
Doesn't this shift the burden of user/group creation to the user though? If the user wants persistent state, they need to have a user/group with predictable UID/GID. With this change, now the user is responsible for arranging this. But maybe this is a non-argument?

Reply:
Now that I think about it, I think the original setup didn't have predictable UID/GID anyways so this might be fine :)